### PR TITLE
Call update_settings_glue even in SIDE_MODULE, so BINARYEN_FEATURES is propagated.

### PR DESCRIPTION
**Pull request by kripken**
Refactor optimize_syscalls into update_settings_glue.

### CODE REVIEW

**Comment by kripken** [comment]
Added a proper test, and enabled a related test as well.

File: tests/test_core.py [hunk 1]
```diff
@@ -59,6 +59,18 @@ def decorated(self):
...
 
+def bleeding_edge_wasm_backend(f):
+  def decorated(self):
+    if not self.is_wasm_backend():
+      self.skipTest('only works in wasm backend')
+    if not V8_ENGINE or V8_ENGINE not in JS_ENGINES:
+      self.skipTest('only works in d8 for now')
+    if self.is_wasm_backend() and not self.get_setting('WASM'):
+      self.skipTest('wasm2js only supports MVP for now')
+    f(self, js_engines=[V8_ENGINE])
```
Comment (tlively) [inline tests/test_core.py]: Do we need to append feature flags to `V8_ENGINE` to support the bleeding edge features, or do we assume the user has already done that in their config?

File: tests/test_core.py [hunk 2]
```diff
@@ -3374,6 +3386,47 @@ def indir(name):
...
+
+    self.prep_dlfcn_lib()
+    lib_src = r'''
+        extern "C" int magic(float x) {
+          return __builtin_wasm_trunc_saturate_s_i32_f32(x);
+        }
+      '''
+    dirname = self.get_dir()
+    filename = os.path.join(dirname, 'liblib.cpp')
+    self.set_setting('EXPORTED_FUNCTIONS', ['_splat_inc_and_get'])
```
Comment (tlively) [inline tests/test_core.py]: Where does `_splat_inc_and_get` come from?

File: tests/test_core.py (hunk 1 above)
Comment (kripken) [reply tests/test_core.py]: I added them to emcc_args manually in the test. Not sure if there's a nicer way? Maybe a parameter to the decorator? Not sure that's clearer.

File: tests/test_core.py (hunk 2 above)
Comment (kripken) [reply tests/test_core.py]: Ah.. that should be "magic" after I renamed it. But the line is not needed anyhow, there isn't a need to export functions in a side module (we don't dce shared modules by default). Will fix, thanks.

File: tests/test_core.py (hunk 1 above)
Comment (tlively) [reply tests/test_core.py]: Yeah actually a decorator parameter that can take flags like `--experimental-wasm-simd` would be good, so you can say just which part of the bleeding edge you care about. I guess this particular test doesn't need that, though.

**Pull request merged**